kuksa_databroker/databroker/src/broker.rs

```diff
@@ -1024,6 +1026,10 @@ impl<'a, 'b> DatabaseWriteAccess<'a, 'b> {
         allowed: Option<types::DataValue>,
         datapoint: Option<Datapoint>,
     ) -> Result<i32, RegistrationError> {
+        if !glob::is_valid_pattern(name.as_str()) {
```

Comment:
### 1. Validate path
This is not correct, as what we want to validate is that a new entry has a valid **path**.

As currently implemented, it will allow `*` and `**` in the path (as `is_valid_pattern(...)` validates a **pattern**).

### 2. Validate pattern
And when `DatabaseWriteAccess::add` has been updated to accept only valid **paths**, we know that any entry in the database has a valid path.

Then, when later constructing "a matcher" that can be used to match paths with a pattern, `is_valid_pattern(...)` can be used to verify that the pattern is valid. But the pattern validation should **not** duplicate the validation of what a valid path can contain (as we already know that all entries we are matching against have a valid path). Instead, it should only validate that the pattern is valid in terms of wildcards, i.e. it should allow any character in a path (which is what this PR is about).

Reply:
is_valid_path method added to split different validations